osu.Game/Screens/Play/HUDOverlay.cs

```diff
@@ -95,8 +99,13 @@ public virtual void BindHitRenderer(HitRenderer hitRenderer)
 
             // in the case a replay isn't loaded, we want some elements to only appear briefly.
             if (!hitRenderer.HasReplayLoaded)
+            {
+                ReplaySettingsOverlay.Hide();
+                ReplaySettingsOverlay.AlwaysPresent = false;

```

Comment:
That's why I suggested putting the key event for toggling visibility in this class' OnKeyDown.